File: code-review-graph/tests/fixtures/sample.php
<?php

namespace App\Models;

use Exception;

interface Repository {
    public function findById(int $id): ?User;
    public function save(User $user): void;
}

class User {
    public int $id;
    public string $name;

    public function __construct(int $id, string $name) {
        $this->id = $id;
        $this->name = $name;
    }

    public function toString(): string {
        return "User({$this->id}, {$this->name})";
    }
}

class InMemoryRepo implements Repository {
    private array $users = [];

    public function findById(int $id): ?User {
        return $this->users[$id] ?? null;
    }

    public function save(User $user): void {
        $this->users[$user->id] = $user;
        echo "Saved " . $user->toString() . "\n";
    }
}

function createUser(Repository $repo, string $name): User {
    $user = new User(count($repo->users ?? []) + 1, $name);
    $repo->save($user);
    return $user;
}

File: code-review-graph/tests/fixtures/Sample.cs
using System;
using System.Collections.Generic;

namespace SampleApp
{
    public interface IRepository
    {
        User FindById(int id);
        void Save(User user);
    }

    public class User
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }

    public class InMemoryRepo : IRepository
    {
        private Dictionary<int, User> _users = new();

        public User FindById(int id)
        {
            return _users.ContainsKey(id) ? _users[id] : null;
        }

        public void Save(User user)
        {
            _users[user.Id] = user;
            Console.WriteLine($"Saved user {user.Id}");
        }
    }

    public class UserService
    {
        private IRepository _repo;

        public UserService(IRepository repo)
        {
            _repo = repo;
        }

        public User GetUser(int id)
        {
            return _repo.FindById(id);
        }
    }
}
